sklearn/linear_model/quantile.py

```diff
@@ -0,0 +1,277 @@
+
+    xtol : float, default 1e-6
+        Global optimization will stop when |w_{t-1} - w_t| < xtol
+        where w_t is result of t'th approximated optimization.
+
+    Attributes
+    ----------
+    coef_ : array, shape (n_features,)
+        Features got by optimizing the Huber loss.
+
+    intercept_ : float
+        Bias.
+
+    n_iter_ : int
+        Number of iterations that scipy.optimize.mimimize has run for.
+
+    References
+    ----------
+    .. [1] Koenker, R., & Bassett Jr, G. (1978). Regression quantiles.
+            Econometrica: journal of the Econometric Society, 33-50.
+
+    .. [2] Chen, C., & Wei, Y. (2005).
+           Computational issues for quantile regression.
+           Sankhya: The Indian Journal of Statistics, 399-417.
+    """
+
+    def __init__(self, quantile=0.5, max_iter=1000, alpha=0.0001, l1_ratio=0.0,
+                 warm_start=False, fit_intercept=True,
+                 gamma=1e-2, gtol=1e-4, xtol=1e-6):
+        self.quantile = quantile
+        self.max_iter = max_iter
+        self.alpha = alpha
+        self.l1_ratio = l1_ratio
+        self.warm_start = warm_start
+        self.fit_intercept = fit_intercept
+        self.gtol = gtol
+        self.xtol = xtol
+        self.gamma = gamma
+
+    def fit(self, X, y, sample_weight=None):
+        """Fit the model according to the given training data.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            Training vector, where n_samples in the number of samples and
+            n_features is the number of features.
+
+        y : array-like, shape (n_samples,)
+            Target vector relative to X.
+
+        sample_weight : array-like, shape (n_samples,)
+            Weight given to each sample.
+
+        Returns
+        -------
+        self : object
+            Returns self.
+        """
+        X, y = check_X_y(
+            X, y, copy=False, accept_sparse=['csr'], y_numeric=True)
+        if sample_weight is not None:
+            sample_weight = np.array(sample_weight)
+            check_consistent_length(y, sample_weight)
+        else:
+            sample_weight = np.ones_like(y)
+
+        if self.quantile >= 1.0 or self.quantile <= 0.0:
+            raise ValueError(
+                "Quantile should be strictly between 0.0 and 1.0, got %f"
+                % self.quantile)
+
+        if self.warm_start and hasattr(self, 'coef_'):
+            parameters = np.concatenate(
+                (self.coef_, [self.intercept_]))
+        else:
+            if self.fit_intercept:
+                parameters = np.zeros(X.shape[1] + 1)
+            else:
+                parameters = np.zeros(X.shape[1] + 0)
+
+        # solve sequence of optimization problems
+        # with different smoothing parameter
+        total_iter = []
+        loss_args = (X, y, self.quantile, self.alpha, self.l1_ratio,
+                     sample_weight)
+        for i in range(10):
```

Comment:
This seems a little strange to have 10 here. Could you explain more?

Reply:
Yes, number of iterations has been a "magic constant": maximal number of cost function detalizations (with decreasing ``gamma``) was hardcoded to 10 . Now it is a class parameter.